Scraper fixes, `ModelListItem` refactor & UI polish

bmartin042503 opened the pull request:
Closes https://github.com/4foureyes/avallamaplanning/issues/70
Closes https://github.com/4foureyes/avallamaplanning/issues/107
Closes https://github.com/4foureyes/avallamaplanning/issues/110

### Changes
- Fixed a bug preventing the Scraper process from stopping correctly when the Cancel button is pressed
- Fixed a bug where the Model Manager failed to automatically apply the "Downloaded" sort order upon initialization
- Fixed a bug where clearing the search text did not restore the full list of models
- Refactored `ModelListItem` into a `TemplatedControl`
- Improved Scraper concurrency for better efficiency
- Simplified several properties to utilize the latest C# 14 features
- Optimized and centralized the rotation animation in `Styles/General.axaml`. It can now be applied to any element using the `spinnerAnimation` class.
- Added a `MaxWidth` to the `SettingsView` for better readability on wide screens and cleaned up unnecessary rows in the layout Grid
- Updated the global `SelectionBrush` value for better visibility
- Minor UI adjustments
- Updated the README (macOS known issues) to include the missing `libFullScreenCheck.dylib` issue

bmartin042503 committed:
scraper cancellation fix + settingsview changes

bmartin042503 committed:
refactored modellistitem + ui changes

bmartin042503 committed:
test case fix + minor adjust

bmartin042503 committed:
remove unnecessary code from ollamaservice

eyeonspringfield commented on avallama/Services/OllamaService.cs:
```diff
@@ -515,37 +474,24 @@ public async IAsyncEnumerable<DownloadResponse> PullModel(string modelName)
-    }
-
-    public async Task<IList<OllamaModelFamily>> GetScrapedFamiliesAsync()
+    public Task<IList<OllamaModelFamily>> GetScrapedFamiliesAsync()
```
Could you add tests to avoid future tech debt?

bmartin042503 replied:
sure

eyeonspringfield commented on avallama/Utilities/Scraper/OllamaLibraryScraper.cs:
```diff
@@ -87,101 +96,112 @@ async IAsyncEnumerable<OllamaModel> StreamModels()
-                var labelSpanNodes = aNode.SelectNodes("div/div/span");
+                    // Popular tags: list of span under div > div > span
+                    // e.g., tags are shown in some nested span set
+                    HtmlNodeCollection? labelSpanNodes = aNode.SelectNodes("div/div/span");
```
unnecessary change from `var` to `HtmlNodeCollection?` IMO

bmartin042503 replied:
I've changed because with var it's making it a non-nullable (`HtmlNodeCollection`) which is a mistake I think, since `SelectNodes` can return null values based on the documentation and how it works:

> An HtmlNodeCollection containing a collection of nodes matching the XPath query, or null if no node matched the XPath expression.

So I wanted to emphasize it that it's nullable and also do checks so a `NullReferenceException` can be avoided, at least I think.

bmartin042503 replied:
but I guess it's not making any difference since I checked it now and it only returns `HtmlNodeCollection?` if .NET 8 is used, otherwise I guess it's returning an empty `HtmlNodeCollection`

bmartin042503 replied:
so i will return to var

eyeonspringfield replied:
I see, weird how this isn't picked up by the compiler

eyeonspringfield replied:
> but I guess it's not making any difference since I checked it now and it only returns `HtmlNodeCollection?` if .NET 8 is used, otherwise I guess it's returning an empty `HtmlNodeCollection`

Oh I see now lol

eyeonspringfield commented on README.md:
```diff
@@ -146,7 +146,7 @@ This is a list of issues we are aware of that will be fixed in a subsequent rele
 
 **macOS**:
-- The application installation process is not ideal, as it requires manual extraction from a ZIP file.
+- The installation process is missing the inclusion of the `native/macos/libFullScreenCheck.dylib` file, which causes the application to launch with a black screen. For Avallama to function correctly, this file must be placed in the `Avallama.app/Contents/macOS` directory.
```
Instead of rewriting the known issue, could you write it as a second one in the list? These 2 issues are separate from each other.

Also better wording suggestion:
> The installed application is missing the `native/macos/libFullScreenCheck.dylib` file. This causes the application to launch with a black screen. A temporary fix is to place this file in the `Avallama.app/Contents/macOS` directory.

bmartin042503 replied:
I rewrote the previous issue with this one on purpose since having to extract the application from a .zip file is totally fine on macOS, many applications work this way (VS Code for example). The dmg way of mounting an app (which I plan to do) and placing it into the Applications folder is just more popular and easier.

And yes I can rewrite it, it sounds better that way.

eyeonspringfield replied:
I see, I thought this way was non-standard, okay.

eyeonspringfield commented on avallama/Styles/AppColors.axaml:
```diff
@@ -5,12 +5,13 @@ Licensed under the MIT License. See LICENSE file for details.
+        <!-- Light theme colors -->
         <ResourceDictionary x:Key="Light">
-            <Color x:Key="Selection">#80755085</Color> <!-- 50% opacity -->
+            <Color x:Key="Selection">#80E4B7F3</Color> <!-- 50% opacity -->
```
Why are the color codes changed?

bmartin042503 replied:
that's the global `SelectionBrush` value which I changed to improve visibility, since the previous one was really similar to the elements it was used on

bmartin042503 committed:
refactor scraper to a service + test cases

bmartin042503 commented:
### Changes

- Refactored the scraper to a service (`OllamaScraperService.cs`) and its associated classes so it can be tested easily
- Added `Microsoft.Extensions.Http` as a dotnet package so the `HttpClient` can be injected efficiently into `OllamaScraperService`
- Added test cases for `OllamaScraperService`
- Rewrite of macOS issue in README

eyeonspringfield reviewed (approved):
lgtm
